consensus/consensus-types/src/pipelined_block.rs

```diff
@@ -286,6 +288,10 @@ impl PipelinedBlock {
             .take()
             .expect("pre_commit_result_rx missing.")
     }
+
+    pub fn set_qc(&self, qc: Option<Arc<QuorumCert>>) {
```

Comment:
why this takes an Option?

Reply:
Removed the option and made an assertion when calling set_qc